pkg/service/controller/servingcert/secret_updating_controller.go

```diff
@@ -161,29 +163,30 @@ func (sc *ServiceServingCertUpdateController) processNextWorkItem() bool {
 // syncSecret will sync the service with the given key.
 // This function is not meant to be invoked concurrently with the same key.
 func (sc *ServiceServingCertUpdateController) syncSecret(key string) error {
-	obj, exists, err := sc.secretCache.GetByKey(key)
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	sharedSecret, err := sc.secretLister.Secrets(namespace).Get(name)
+	if kapierrors.IsNotFound(err) {
+		glog.V(4).Infof("Secret has been deleted %v", key)
+		return nil
+	}
 	if err != nil {
 		glog.V(4).Infof("Unable to retrieve service %v from store: %v", key, err)
```

Comment:
this will double log. Remove this printf

Reply:
the above return

Reply:
> the above return

it double logs because the main handler logs errors that are returned already.